### PR TITLE
feat: Refactor sdk.ts

### PR DESCRIPTION
**Pull request by chrisduma-ledger**
Refactor SDK file

### CODE REVIEW

**Review by sarneijim — CHANGES_REQUESTED** [review]
Please, avoid code duplication.
If you want to change the name must be "exchange003" instead, and if you are going to change it you must change it across all the file.

**Review by CremaFR — APPROVED** [review]

**Comment by chrisduma-ledger** [comment]
@sarneijim 
> and if you are going to change it you must change it across all the file.

As I have written in Slack, I will add it for the whole file if the teams agree with the proposal 
Screenshot 2024-09-17 at 13 43 11

> If you want to change the name must be "exchange003" instead

If we use `exchange` as a name, then we will not be able to differentiate between apps, and the swap team needs `003` to not display, we want it to display, please make a proposal for the issue.

File: lib/src/error.ts [hunk 1]
```diff
@@ -1,3 +1,6 @@
+import { ExchangeType } from "@ledgerhq/wallet-api-client";
+import { ErrorOrigin } from "./sdk";
```
Comment (sprohaszka-ledger) [inline lib/src/error.ts:2]: This dependency create a cycle one between `error.ts` and `sdk.ts`.
We can either replace the parameter as a string or if you want to enforce the use of an enum, extract this enum definition in a different file.

File: lib/src/sdk.ts [hunk 2]
```diff
@@ -98,18 +89,22 @@ enum FeeStrategyEnum {
...
 
-// Should be available from the WalletAPI (zod :( )
-const ExchangeType = {
+export const ExchangeType = {
   FUND: "FUND",
   SELL: "SELL",
   SWAP: "SWAP",
 } as const;
 
+export enum ErrorOrigin {
```
Comment (sprohaszka-ledger) [inline lib/src/sdk.ts:98]: Why don't reuse `ExchangeType`? It seems they both serve the same purpose (characterize exchange transaction).

File: lib/src/sdk.ts [hunk 3]
```diff
@@ -98,18 +89,22 @@ enum FeeStrategyEnum {
...
+  "sell",
+  "swap",
+  "fund",
+}
+
 /**
- * ExchangeSDK allows you to send a swap request to Ledger Device, through a Ledger Live request.
- * Under the hood it relies on {@link https://github.com/LedgerHQ/wallet-api WalletAPI}.
+ * ExchangeSDK allows you to send a swap request to a Ledger Device through a Ledger Live request.
+ * Under the hood, it relies on the WalletAPI.
```
Comment (sprohaszka-ledger) [inline lib/src/sdk.ts:106]: The link didn't work?

**Review by sprohaszka-ledger — COMMENTED** [review]
A few comments. The one that bother me most is the dependency one.

**Review by sprohaszka-ledger — APPROVED** [review]
LGTM

**Review by sarneijim — APPROVED** [review]
THANK YOU!

**Pull request merged**